### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'request'

### ISSUE THREAD

**Issue by corporatecoder** [adafruit/Adafruit_CircuitPython_Requests#122]
Tried running the following in REPL but it gave an error.
```
>>> import adafruit_requests
>>> r = adafruit_requests.get('https://www.google.com')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "adafruit_requests.py", line 809, in get
AttributeError: 'NoneType' object has no attribute 'request'
```

I looked at the adafruit_requests.py file and it appears that in line 735 `_default_session` is defined as `None`. Thus, when `_default_session.request("GET", url, **kw)` is called in line 809, this error arises. Am I wrong to use this library as I would with basic Python?

**Comment by anecdata** [comment]
Is this `espressif`?

```py
import wifi
import socketpool
import ssl
import adafruit_requests

# local
from secrets import secrets

URL = "https://example.com"

wifi.radio.connect(secrets['ssid'], secrets['password'])

pool = socketpool.SocketPool(wifi.radio)
context = ssl.create_default_context()
requests = adafruit_requests.Session(pool, context)

with requests.get(URL) as response:
    print(response.status_code, response.reason, response.headers)
    print(response.content)  # or response.text or response.json()```

**Comment by corporatecoder** [comment]
No I am using an RPi Pico W.

I figured it out. Thanks for the example code. A few notes on issues I ran into for those that may see this.
1.) Check that you specify an ssid and password. I'm new to microcontrollers and overlooked this.
2.) At first I thought secrets was a module and it is, but you actually save the connection information to a file called secrets.py and are importing it. See [here ](https://learn.adafruit.com/electronic-history-of-the-day-with-pyportal/code-walkthrough-secrets-py)
3.) I was using CircuitPython 7.3.3 and was getting import errors on wifi and socketpool. Upgrading to the newest CircuitPython (8.0.0-beta.4) fixed this.
